File: apps/web/types/execution-event.ts
/** ExecutionEvent types matching the backend ExecutionEventRead schema. */

export type EventType =
  | "task_claimed"
  | "task_completed"
  | "task_failed"
  | "artifact_created"
  | "approval_requested"
  | "approval_resolved"
  | "run_started"
  | "run_completed"
  | "run_failed"
  | "plan_generated";

export interface ExecutionEvent {
  id: string;
  event_type: EventType;
  summary: string;
  metadata_: Record<string, unknown> | null;
  project_id: string | null;
  run_id: string | null;
  task_id: string | null;
  artifact_id: string | null;
  agent_slug: string | null;
  created_at: string;
}

export interface ExecutionEventList {
  items: ExecutionEvent[];
  total: number;
}

File: apps/api/app/schemas/execution_event.py
"""Execution event schemas — response models for the event log."""

import uuid
from datetime import datetime

from pydantic import BaseModel

from app.models.execution_event import EventType


class ExecutionEventRead(BaseModel):
    id: uuid.UUID
    event_type: EventType
    summary: str
    metadata_: dict | None
    project_id: uuid.UUID | None
    run_id: uuid.UUID | None
    task_id: uuid.UUID | None
    artifact_id: uuid.UUID | None
    agent_slug: str | None
    created_at: datetime

    model_config = {"from_attributes": True}


class ExecutionEventList(BaseModel):
    items: list[ExecutionEventRead]
    total: int
